src/praisonai-bot/praisonai_bot/gateway/server.py

```
@@ -906,6 +906,16 @@ def __init__(
         # visible in ``health()`` as ``degraded`` instead of vanishing — a
         # skipped channel must be distinguishable from one never configured.
         self._degraded_channels: Dict[str, str] = {}  # channel_name -> reason
+        # Issue #3518: shared, cross-owner degraded-capability registry so
+        # provider/model auth, SecretRef resolution, and MCP capability owners
+        # can record degradation that ``health()`` aggregates into a single
+        # ``degraded_owners`` surface alongside channels. Kept optional/lazy so
+        # gateways that never touch it are unaffected.
+        try:
+            from praisonaiagents.gateway import DegradedCapabilityRegistry
+            self._degraded_registry = DegradedCapabilityRegistry()
+        except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)